provider/core/src/data_provider.rs

```diff
@@ -27,6 +27,7 @@ use icu_locid::LanguageIdentifier;
 pub struct DataRequestMetadata;
 
 #[derive(Default, Debug, PartialEq, Eq, PartialOrd, Ord)]
+#[allow(clippy::exhaustive_structs)] // this type is stable
```

Comment:
Nit: "this type is stable", "stable type", "stable"?

Reply:
yeah we're a bit inconsistent across the codebase about this